app/models/site_status.rb

```diff
@@ -19,6 +19,21 @@ class SiteStatus < ApplicationRecord
 
   after_initialize :set_defaults
   before_validation :set_vac_status
+  before_create :set_new_status
+
+  def set_new_status
+    if !course.site_statuses.empty? && !course.site_statuses.all?(&:status_new_status?)
+      self.status = :running
+    end
+  end
+
+  def destroy
+    if course.site_statuses.any?(&:status_new_status?)
```

Comment:
@tvararu Is this logic correct for mixed courses? I notice that the spec for mixed courses only covers adding new locations, not deleting them.

Reply:
Good spot... I was going to add one but I got distracted and forgot. Will rectify 👍